lib/kubernetes-deploy/container_logs.rb

```diff
@@ -62,16 +62,23 @@ def rfc3339_timestamp(time)
 
     def deduplicate(logs)
       deduped = []
-      check_for_duplicate = true
 
-      logs.each do |line|
-        timestamp, msg = split_timestamped_line(line)
-        next if check_for_duplicate && likely_duplicate?(timestamp)
-        check_for_duplicate = false # logs are ordered, so once we've seen a new one, assume all subsequent logs are new
+      parsed_lines = logs.map { |line| split_timestamped_line(line) }
+      sorted_lines = parsed_lines.sort do |a, b|
+        if a[0].nil? || b[0].nil?
+          0
```

Comment:
Hm. Not sure I'm following this logic. `0` here means the two elements are treated as equal, right? Why is that what we want? I would have expected something like
```ruby
if a[0].nil?
  -1
elsif b[0].nil?
  +1
else
  a[0] <=> b[0]
end
```
or maybe something like
```ruby
(a[0] || 0) <=> (b[0] || 0)
```

Reply:
My thought was that considering them equal was equivalent to leaving them alone, but that's completely wrong. It totally messes up the sort. Since the actual known case where the timestamp is missing is one where that defective line is the first, I think the second suggestion (assuming no timestamp means earliest) is a good approach.